deltachat-ffi/src/lib.rs

```diff
@@ -1780,7 +1786,7 @@ pub unsafe extern "C" fn dc_chat_get_profile_image(chat: *mut dc_chat_t) -> *mut
     let chat = &*chat;
 
     match chat.get_profile_image() {
-        Some(i) => i.strdup(),
+        Some(p) => p.to_str().unwrap().to_string().strdup(),
```

Comment:
`p.to_c_string().unwrap().strdup()` should be possible afaik and will not panic as often.  You may need to `use dc_tools::OsStrExt` for the `.to_c_string()` part and `use dc_tools::StrExt` for the `.strdup()` part.

Same for contact